src/core/raster/qgsrasterlayerrenderer.cpp

```diff
@@ -127,55 +127,78 @@ QgsRasterLayerRenderer::QgsRasterLayerRenderer( QgsRasterLayer *layer, QgsRender
       // layer CRS, then this dummy extent is returned by QgsMapRendererJob::reprojectToLayerExtent()
       // Don't try to reproject it now to view extent as this would return
       // a null rectangle.
-      myProjectedViewExtent = rendererContext.extent();
+      viewExtentInMapCrs = rendererContext.extent();
     }
     else
     {
       try
       {
         QgsCoordinateTransform ct = rendererContext.coordinateTransform();
         ct.setBallparkTransformsAreAppropriate( true );
-        myProjectedViewExtent = ct.transformBoundingBox( rendererContext.extent() );
+        viewExtentInMapCrs = rendererContext.mapExtent(); //ct.transformBoundingBox( rendererContext.extent() );
       }
       catch ( QgsCsException &cs )
       {
         QgsMessageLog::logMessage( QObject::tr( "Could not reproject view extent: %1" ).arg( cs.what() ), QObject::tr( "Raster" ) );
-        myProjectedViewExtent.setNull();
+        viewExtentInMapCrs.setNull();
       }
     }
 
     try
     {
       QgsCoordinateTransform ct = rendererContext.coordinateTransform();
       ct.setBallparkTransformsAreAppropriate( true );
-      myProjectedLayerExtent = ct.transformBoundingBox( layer->extent() );
+      layerExtentInMapCrs = ct.transformBoundingBox( layer->extent() );
     }
     catch ( QgsCsException &cs )
     {
       QgsMessageLog::logMessage( QObject::tr( "Could not reproject layer extent: %1" ).arg( cs.what() ), QObject::tr( "Raster" ) );
-      myProjectedLayerExtent.setNull();
+      layerExtentInMapCrs.setNull();
     }
   }
   else
   {
     QgsDebugMsgLevel( QStringLiteral( "coordinateTransform not set" ), 4 );
-    myProjectedViewExtent = rendererContext.extent();
-    myProjectedLayerExtent = layer->extent();
+    viewExtentInMapCrs = rendererContext.extent();
+    layerExtentInMapCrs = layer->extent();
   }
 
   // clip raster extent to view extent
-  QgsRectangle myRasterExtent = layer->ignoreExtents() ? myProjectedViewExtent : myProjectedViewExtent.intersect( myProjectedLayerExtent );
-  if ( myRasterExtent.isEmpty() )
+  QgsRectangle visibleExtentOfRasterInMapCrs = layer->ignoreExtents() ? viewExtentInMapCrs : viewExtentInMapCrs.intersect( layerExtentInMapCrs );
+  if ( visibleExtentOfRasterInMapCrs.isEmpty() && !rendererContext.coordinateTransform().isShortCircuited() )
```

Comment:
shouldn't we return when isEmpty() && isShortCircuited() ?

Reply:
Indeed, fixed